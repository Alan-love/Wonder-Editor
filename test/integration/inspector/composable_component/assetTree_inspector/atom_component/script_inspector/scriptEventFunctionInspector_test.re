open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("script event function inspector", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());

      MainEditorSceneTool.prepareScene(sandbox);
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("set event function data", () => {
      test("default data is empty closure function which returm {}", () => {
        let assetTreeData =
          MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();

        let addedNodeId = MainEditorAssetIdTool.getNewAssetId();
        MainEditorAssetHeaderOperateNodeTool.addScriptEventFunction();

        ScriptEventFunctionInspectorTool.getEventFunctionDataJsObjStr(
          addedNodeId,
        )
        |> StateLogicService.getEditorState
        |> expect
        == ScriptEventFunctionInspectorTool.buildEventFunctionDataJsObjStr();
      });
      test("test set init,dispose event function", () => {
        let assetTreeData =
          MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();

        let addedNodeId = MainEditorAssetIdTool.getNewAssetId();
        MainEditorAssetHeaderOperateNodeTool.addScriptEventFunction();
        let editorState = StateEditorService.getState();
        let jsObjStr =
          ScriptEventFunctionInspectorTool.buildEventFunctionDataJsObjStr(
            ~initFunc=Some((. script, api, state) => state),
            ~disposeFunc=Some((. script, api, state) => state),
            (),
          );
        ScriptEventFunctionInspectorTool.updateEventFunctionData(
          addedNodeId,
          ScriptEventFunctionInspectorTool.getEventFunctionName(
            addedNodeId,
            editorState,
          ),
          jsObjStr,
        );

        ScriptEventFunctionInspectorTool.getEventFunctionDataJsObjStr(
          addedNodeId,
        )
        |> StateLogicService.getEditorState
        |> expect == jsObjStr;
      });
    });
  });