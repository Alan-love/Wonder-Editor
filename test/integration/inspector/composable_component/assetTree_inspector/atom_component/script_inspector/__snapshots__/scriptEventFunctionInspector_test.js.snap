// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`script event function inspector test snapshot test 1`] = `
<article
  className="wonder-scriptEventFunction-inspector"
>
  <h1>
    ScriptEventFunction
  </h1>
  <hr />
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
      title="script eventFunction's name"
    >
      <span
        className="component-label"
      >
        Name
      </span>
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="New Script EventFunction"
      />
    </div>
  </article>
  <div
    className="scriptEventFunction-data"
  >
    <article
      className="wonder-file-input"
    >
      <button
        onClick={[Function]}
      >
        Set EventFunction Data
      </button>
    </article>
  </div>
</article>
`;
